File: frontend/client/lib/models/geofence.dart
class Geofence {
  final int geofenceId;
  final int poiId;
  final double latitude;
  final double longitude;
  final double radius;

  Geofence({
    required this.geofenceId,
    required this.poiId,
    required this.latitude,
    required this.longitude,
    required this.radius,
  });

  factory Geofence.fromJson(Map<String, dynamic> json) {
    return Geofence(
      geofenceId: json['geofenceId'],
      poiId: json['poiId'],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      radius: (json['radius'] as num).toDouble(),
    );
  }
}

File: frontend_fix/client/lib/model/Geofence.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace client.lib.model
{
    public class Geofence
    {
        [JsonPropertyName("geofenceId")]
        public int GeofenceId { get; set; }

        [JsonPropertyName("poiId")]
        public int PoiId { get; set; }

        [JsonPropertyName("latitude")]
        public double Latitude { get; set; }

        [JsonPropertyName("longitude")]
        public double Longitude { get; set; }

        [JsonPropertyName("radius")]
        public double Radius { get; set; }
    }
}
